### PR TITLE
Feature: Add basic CI

### PR DESCRIPTION
**Pull request by vcwild**
## What is being done?

- Adding basic CI to the project
- Changes the project structure one folder up
- Updates gitignore with C and python project
- Adds minunit header

Depends on #2 
Closes #3 #4

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        run: |
+          pip install -U pip
+          pip install norminette
+          norminette sources/ includes/
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Compile the program
+        run: make
```
Comment (iaurg) [inline .github/workflows/ci.yml]: Maybe we need to point exact folder to run make?

Comment (vcwild) [reply .github/workflows/ci.yml]: Yes, the current design is using a subfolder to carry the entire project...

Comment (vcwild) [reply .github/workflows/ci.yml]: it might be a good idea for automation to remove the subfolder

Comment (iaurg) [reply .github/workflows/ci.yml]: I can change it, what do you think?

Comment (vcwild) [reply .github/workflows/ci.yml]: It's already done in the MR

**Comment by vcwild** [comment]
Moving this to draft, this will add some problems for our current design

**Comment by vcwild** [comment]
I will need help testing this. 
It seems the worker instance is unable to use the `libreadline-dev` package for Ubuntu.
I cannot test it by myself, I am using a Fedora machine.

## EDIT:

- Leave this for a later conversation

**Comment by vcwild** [comment]
Please merge this using squash and merge

**Pull request merged**